### ISSUE TITLE
Clean up the main plugin file

### ISSUE THREAD

**Issue by jdevalk** [equalizedigital/accessibility-checker#403]
The main plugin file has far too many things happening and a lot of that doesn't need to be loaded on each request.

## Move out the AJAX functions
All the AJAX functions that are currently in the main plugin file should be offloaded to a separate class. That class needs to only be instantiated when the current request is actually an AJAX request, like so:

```php
if ( defined( 'DOING_AJAX' ) && DOING_AJAX ) {
  $ajax = new EDAC\Ajax;
  $ajax->init_hooks();
}
```

Then all the functions could be moved into that class and renamed for a lot more simplicity. Instead of:

```php
add_action( 'wp_ajax_edac_summary_ajax', 'edac_summary_ajax' );
```

You'd have:

```php
add_action( 'wp_ajax_edac_summary_ajax', [ $this, 'summary' ] );
```
## Move out the frontend functionality
A separate class should be made for all the frontend functions, and that's the only thing that needs to be loaded on the frontend when the current user is not logged in.

Literally only these two hooks as far as I can tell:

```php
add_filter( 'the_content', 'edac_output_simplified_summary' );
add_action( 'wp_footer', 'edac_output_accessibility_statement' );
```

## Move the admin only functionality to admin classes
It should probably be more than one file with a bit of logical grouping, but it needs to be split up for maintainability and simply because it doesn't all have to be loaded all the time.

**Comment by SteveJonesDev** [comment]
Fixed with release: https://github.com/equalizedigital/accessibility-checker/releases/tag/v1.8.0
